src/zeroband/inference/monkeypatch_sampling_metadata.py

```diff
@@ -0,0 +1,175 @@
+from typing import Optional
+
+import torch
+from vllm.model_executor.sampling_metadata import SamplingMetadataCache, SequenceGroupToSample
+from vllm.sampling_params import SamplingType
+from vllm.sequence import SequenceGroupMetadata
+
+
+def fixed_prepare_seq_groups(
+    seq_group_metadata_list: list[SequenceGroupMetadata],
+    seq_lens: list[int],
+    query_lens: list[int],
+    device: str,
+    generators: Optional[dict[str, torch.Generator]] = None,
+    cache: Optional[SamplingMetadataCache] = None,
```

Comment:
```suggestion
    generators: dict[str, torch.Generator] | None = None,
    cache: dict[SamplingMetadataCache] | None = None,
```

Reply:
hrmm I think keeping it as similar as possible to the way it is defined in vllm is probably better. this code wont live too long so should be fine